File: src/julia/kamada_kawai/kamada_kawai2csv.jl
module Kamada_Kawai2csv

using ..Kamada_Kawai

export writeState
function writeState(fp::IOStream, state::Vector{Coord})
    for i in 1:length(state)
        write(fp, string(state[i].x))
        write(fp, ",")
        write(fp, string(state[i].y))
        write(fp, "\n")
    end
    write(fp, "\n")
end # writeState

end # module

File: temp/kk/memory/c/kamada_kawai/kamada_kawai2csv.c
#include <stdio.h>
#include <stdlib.h>
#include <json-c/json.h>

#include "./kamada_kawai2csv.h"

void writeState(FILE *fp, Coord *state, int n)
{
    for (int i = 0; i < n; i++)
    {
        fprintf(fp, "%f,%f\n", state[i].x, state[i].y);
    }
    fprintf(fp, "\n");
}